ZeeHomeScreen/Classes/Components/BasicContainerCell/UniversalCollectionViewCell.m

```diff
@@ -6,76 +6,45 @@
 //
 
 #import "UniversalCollectionViewCell.h"
-#import "CarouselViewController.h"
-#import <ZeeHomeScreen/ComponenttFactory.h>
 #import <ZeeHomeScreen/ZeeHomeScreen-Swift.h>
 
 @interface UniversalCollectionViewCell()
 
-@property (nonatomic, strong) NSLayoutConstraint *cellWidthConstraint;
-@property (nonatomic, strong) NSLayoutConstraint *cellHeightConstraint;
+@property (nonatomic) NSLayoutConstraint *cellWidthConstraint;
+@property (nonatomic) NSLayoutConstraint *cellHeightConstraint;
 
 - (void)updateFlexibilityConstraints:(ComponentModel *)componentModel;
 
 @end
 
 @implementation UniversalCollectionViewCell
 
-- (instancetype)initWithFrame:(CGRect)frame {
-    self = [super initWithFrame:frame];
-    if (self) {
-    
-    }
-    return self;
-}
-
 -(void)prepareForReuse {
     [super prepareForReuse];
-    
-    if ([self.componentViewController respondsToSelector:@selector(prepareComponentForReuse)]) {
-        [self.componentViewController prepareComponentForReuse];
+
+    if (self.parentHandlesReusingComponent) {
+        if ([self.componentViewController respondsToSelector:@selector(prepareComponentForReuse)]) {
+            [self.componentViewController prepareComponentForReuse];
+        }
+    } else {
+        [self.componentViewController removeViewFromParentViewController];
```

Comment:
when parent won't handle reusing the component, just remove it when cell goes out of bounds, preventing memory from leaking